tests/wasmfs/wasmfs_fetch.c

```diff
@@ -17,12 +18,92 @@
 #include <sys/stat.h>
 #include <unistd.h>
 
-static backend_t make_js_file_backend(void* arg) {
-  return wasmfs_create_js_file_backend();
+char url_orig[1024];
+void set_url_origin() {
+  url_orig[0] = 0;
+  EM_ASM(
```

Comment:
We should try to avoid `EM_ASM` since it has a performance cost. It's better to import functions directly from the JS library by writing C function declarations for them.

Reply:
Moved to a separate file.